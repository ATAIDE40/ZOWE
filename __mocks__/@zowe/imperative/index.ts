/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Mocks all imperative classes specified under __mocks__/__imperative__
 * @returns mocked version of the @zowe/imperative module
 */
const __imperative__get = function () {
  const fs = require.requireActual("fs");
  const imperativePath = require("path").resolve(__dirname, "..", "..", "__imperative__");
  const extraClasses: any = {};
  fs.readdirSync(imperativePath).forEach((file: string) => {
    extraClasses[file.split(".")[0]] = require(`${imperativePath}/${file}`);
  });
  return {
    ...require.requireActual("@zowe/imperative"),
    // Note: Comment out the line below if you do not want to mock
    //       the classes in __mocks__/__imperative__/ for local testing purposes
    ...extraClasses
    // Note: You can still use Imperative.console.info("debug") in your test
  };
}

module.exports = __imperative__get();
