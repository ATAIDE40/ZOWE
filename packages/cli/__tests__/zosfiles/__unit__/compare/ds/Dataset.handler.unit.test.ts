/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { Get } from "@zowe/zos-files-for-zowe-sdk";
import { UNIT_TEST_ZOSMF_PROF_OPTS } from "../../../../../../../__tests__/__src__/mocks/ZosmfProfileMock";

describe("Compare data set handler", () => {
    describe("process method", () => {
        it("should compare two data sets", async () => {
            // Require the handler and create a new instance
            const handlerReq = require("../../../../../src/zosfiles/compare/ds/Dataset.handler");
            const handler = new handlerReq.default();
            const dataSetName1 = "testing1";
            const dataSetName2 = "testing2";
            const options = "--ns";
            const binary = true;


            // Vars populated by the mocked function
            let error;
            let apiMessage = "";
            let jsonObj;
            let logMessage = "";
            let fakeSession = null;

            // Mock the compare ds function
            Get.dataSet = jest.fn((session) => {
                fakeSession = session;
                return {
                    success: true,
                    commandResponse: "compared"
                };
            });

            // Mocked function references
            const profFunc = jest.fn((args) => {
                return {
                    host: "fake",
                    port: "fake",
                    user: "fake",
                    password: "fake",
                    auth: "fake",
                    rejectUnauthorized: "fake",
                };
            });

            try {
                // Invoke the handler with a full set of mocked arguments and response functions
                await handler.process({
                    arguments: {
                        $0: "fake",
                        _: ["fake"],
                        dataSetName1,
                        dataSetName2,
                        options,
                        ...UNIT_TEST_ZOSMF_PROF_OPTS
                    },
                    response: {
                        data: {
                            setMessage: jest.fn((setMsgArgs) => {
                                apiMessage = setMsgArgs;
                            }),
                            setObj: jest.fn((setObjArgs) => {
                                jsonObj = setObjArgs;
                            })
                        },
                        console: {
                            log: jest.fn((logArgs) => {
                                logMessage += "\n" + logArgs;
                            })
                        },
                        progress: {
                            startBar: jest.fn((parms) => {
                                // do nothing
                            }),
                            endBar: jest.fn(() => {
                                // do nothing
                            })
                        }
                    },
                    profiles: {
                        get: profFunc
                    }
                } as any);
            } catch (e) {
                error = e;
            }

            expect(Get.dataSet).toHaveBeenCalledTimes(2);
            expect(Get.dataSet).toHaveBeenCalledWith(fakeSession, dataSetName1, {
                task: {
                    percentComplete: 0,
                    stageName: 0,
                    statusMessage: "Retrieving second dataset"
                }
            });
            expect(jsonObj).toMatchSnapshot();
            expect(apiMessage).toMatchSnapshot();
            expect(logMessage).toMatchSnapshot();
        });
    });
});
