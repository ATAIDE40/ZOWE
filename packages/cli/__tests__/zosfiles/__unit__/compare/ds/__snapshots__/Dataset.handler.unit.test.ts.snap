// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare data set handler process method should compare two data sets 1`] = `
Object {
  "apiResponse": Object {},
  "commandResponse": "[2mCompared values have no visual difference.[22m",
  "success": true,
}
`;

exports[`Compare data set handler process method should compare two data sets 2`] = `""`;

exports[`Compare data set handler process method should compare two data sets 3`] = `
"
[2mCompared values have no visual difference.[22m"
`;
